builtin/logical/transit/path_rewrap.go

```diff
@@ -45,94 +67,146 @@ func (b *backend) pathRewrap() *framework.Path {
+		batchInputItems[0] = BatchRequestItem{
+			Ciphertext: ciphertext,
+		}
+
+		// Decode the context
+		contextRaw := d.Get("context").(string)
+		if len(contextRaw) != 0 {
+			batchInputItems[0].Context, err = base64.StdEncoding.DecodeString(contextRaw)
+			if err != nil {
+				return logical.ErrorResponse("failed to base64-decode context"), logical.ErrInvalidRequest
+			}
+		}
+
+		// Decode the nonce
+		nonceRaw := d.Get("nonce").(string)
+		if len(nonceRaw) != 0 {
+			batchInputItems[0].Nonce, err = base64.StdEncoding.DecodeString(nonceRaw)
+			if err != nil {
+				return logical.ErrorResponse("failed to base64-decode nonce"), logical.ErrInvalidRequest
+			}
 		}
 	}
 
-	// Decode the nonce if any
-	nonceRaw := d.Get("nonce").(string)
-	var nonce []byte
-	if len(nonceRaw) != 0 {
-		nonce, err = base64.StdEncoding.DecodeString(nonceRaw)
-		if err != nil {
-			return logical.ErrorResponse("failed to base64-decode nonce"), logical.ErrInvalidRequest
+	batchResponseItems := make([]BatchResponseItem, len(batchInputItems))
+	contextSet := len(batchInputItems[0].Context) != 0
+
+	for i, item := range batchInputItems {
+		if (len(item.Context) == 0 && contextSet) || (len(item.Context) != 0 && !contextSet) {
+			return logical.ErrorResponse("context should be set either in all the request blocks or in none"), logical.ErrInvalidRequest
+		}
+
+		if item.Ciphertext == "" {
+			batchResponseItems[i].Error = "missing ciphertext to decrypt"
+			continue
 		}
 	}
 
 	// Get the policy
-	p, lock, err := b.lm.GetPolicyShared(req.Storage, name)
+	p, lock, err := b.lm.GetPolicyShared(req.Storage, d.Get("name").(string))
 	if lock != nil {
 		defer lock.RUnlock()
 	}
 	if err != nil {
 		return nil, err
 	}
-	// Error if invalid policy
 	if p == nil {
 		return logical.ErrorResponse("policy not found"), logical.ErrInvalidRequest
 	}
 
-	plaintext, err := p.Decrypt(context, nonce, value)
-	if err != nil {
-		switch err.(type) {
-		case errutil.UserError:
-			return logical.ErrorResponse(err.Error()), logical.ErrInvalidRequest
-		case errutil.InternalError:
-			return nil, err
-		default:
-			return nil, err
+	for i, item := range batchInputItems {
+		if batchResponseItems[i].Error != "" {
+			continue
 		}
-	}
 
-	if plaintext == "" {
-		return nil, fmt.Errorf("empty plaintext returned during rewrap")
-	}
+		plaintext, err := p.Decrypt(item.Context, item.Nonce, item.Ciphertext)
+		if err != nil {
+			switch err.(type) {
+			case errutil.UserError:
+				batchResponseItems[i].Error = err.Error()
+				continue
+			default:
+				return nil, err
+			}
+		}
 
-	ciphertext, err := p.Encrypt(context, nonce, plaintext)
-	if err != nil {
-		switch err.(type) {
-		case errutil.UserError:
-			return logical.ErrorResponse(err.Error()), logical.ErrInvalidRequest
-		case errutil.InternalError:
-			return nil, err
-		default:
-			return nil, err
+		if plaintext == "" {
+			batchResponseItems[i].Error = "empty plaintext returned during rewrap"
```

Comment:
@jefferai This could be one reason why the response should not be a logical.ErrorResponse.